gio/src/inet_socket_address.rs

```diff
@@ -0,0 +1,24 @@
+use InetAddress;
+use InetSocketAddress;
+use InetSocketAddressExt;
+
+use std::net::SocketAddr;
+
+impl InetSocketAddress {
+    pub fn from_socket_address(addr: SocketAddr) -> Self {
+        InetSocketAddress::new::<InetAddress>(&addr.ip().into(), addr.port())
+    }
+}
+
+impl From<SocketAddr> for InetSocketAddress {
+    fn from(addr: SocketAddr) -> Self {
+        InetSocketAddress::from_socket_address(addr)
+    }
+}
+
+impl Into<SocketAddr> for InetSocketAddress {
+    fn into(self) -> SocketAddr {
+        //FIXME: unwrap is unsound here, may use TryInto?
+        SocketAddr::new(self.get_address().unwrap().into(), self.get_port())
```

Comment:
No, that's OK for now. This can't possibly return `None` and I'm going to fix that once https://gitlab.gnome.org/GNOME/glib/-/merge_requests/1708 is merged into gio.

Reply:
Ah Okay :).